Compilation fails if json-c is not enabled

If I set up meson with "--force-fallback-for=libnvme .build" i get a build error:

```
[67/93] Compiling C object nvme.p/nvme-print.c.o
FAILED: nvme.p/nvme-print.c.o 
cc -Invme.p -I. -I.. -Iccan -I../ccan -Isubprojects/libnvme/src -I../subprojects/libnvme/src -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -g -fomit-frame-pointer -D_GNU_SOURCE -include config.h -DCCAN_LIST_DEBUG=1 -DCCAN_STR_DEBUG=1 -MD -MQ nvme.p/nvme-print.c.o -MF nvme.p/nvme-print.c.o.d -o nvme.p/nvme-print.c.o -c ../nvme-print.c
In file included from ../nvme-print.c:13:
../nvme-print-json.h:143:33: error: expected ',' or ')', found "char"
  143 | #define json_output_error(const char *msg, va_list ap)
      |                                 ^~~~
```

Worked around this by adding json-c to the force fallback list: (meson setup --force-fallback-for=libnvme,json-c .build)
Maybe json-c should be added to the required dependencies in the README?
The dependency seems pretty important anyway  as plugins depend on it.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -351,8 +351,10 @@ jobs:
           muon setup                    \
              -Dprefix=${ROOT_PATH}      \
              -Dwrap_mode=forcefallback  \
+             -Dlibnvme:json-c=disabled  \
              -Dlibnvme:python=disabled  \
              -Dlibnvme:openssl=disabled \
+             -Djson-c=disabled          \
              .build
           samu -C .build
           muon -C .build install

--- a/nbft.c
+++ b/nbft.c
@@ -18,6 +18,7 @@ static void print_connect_msg(nvme_ctrl_t c)
 
 static void json_connect_msg(nvme_ctrl_t c)
 {
+#ifdef CONFIG_JSONC
 	struct json_object *root;
 
 	root = json_create_object();
@@ -26,6 +27,7 @@ static void json_connect_msg(nvme_ctrl_t c)
 	json_print_object(root, NULL);
 	printf("\n");
 	json_free_object(root);
+#endif
 }
 
 int nbft_filter(const struct dirent *dent)

--- a/nbft.c
+++ b/nbft.c
@@ -10,6 +10,8 @@
 #include "libnvme.h"
 #include "fabrics.h"
 
+#include "util/types.h"
+
 #define NBFT_SYSFS_FILENAME	"NBFT*"
 
 static void print_connect_msg(nvme_ctrl_t c)

--- a/nbft.c
+++ b/nbft.c
@@ -3,6 +3,7 @@
 #include <errno.h>
 #include <stdio.h>
 #include <fnmatch.h>
+#include <stdlib.h>
 
 #include "nvme.h"
 #include "nbft.h"

--- a/nvme-print-json.h
+++ b/nvme-print-json.h
@@ -142,10 +142,11 @@ void json_connect_msg(nvme_ctrl_t c);
 #define json_persistent_event_log(pevent_log_info, size)
 #define json_predictable_latency_event_agg_log(pea_log, log_entries)
 #define json_predictable_latency_per_nvmset(plpns_log, nvmset_id)
+#define json_output_error(msg, ap)
+#define json_output_result(msg, ap)
 #define json_output_status(status)
-#define json_output_error(const char *msg, va_list ap)
-#define json_output_perror(const char *msg)
-#define json_output_result(const char *msg, va_list ap)
+#define json_output_message(error, msg, ap)
+#define json_output_perror(msg)
 
 /* fabrics.c */
 #define json_discovery_log(log, numrec)

